core/src/main/java/org/arquillian/smart/testing/configuration/ConfigurationReader.java

```diff
@@ -88,4 +91,27 @@ private Path getDefaultConfigFile(File... files) {
             .findFirst()
             .get();
     }
+
+    private Map<String, Object> readEffectiveConfig(Path filePath){
+        Map<String, Object> config = getConfigParametersFromFile(filePath);
+
+        List<Map<String, Object>> configs = new ArrayList<>();
+        configs.add(config);
+        while (config.get(INHERIT) != null) {
+            String inherit = String.valueOf(config.get(INHERIT));
+            filePath = filePath.getParent().resolve(inherit);
+            config = getConfigParametersFromFile(filePath);
+            if (!config.isEmpty()) {
+                configs.add(0, config);
```

Comment:
Alternatively, you can use [`Deque`](https://docs.oracle.com/javase/7/docs/api/java/util/Deque.html) as a Stack - might make it a bit easier to understand what we really do with inheritance.